src/olympia/editors/views.py

```diff
@@ -725,19 +725,33 @@ def review_viewing(request):
     current_name = ''
     is_user = 0
     key = '%s:review_viewing:%s' % (settings.CACHE_PREFIX, addon_id)
+    user_key = '%s:review_viewing_user:%s' % (settings.CACHE_PREFIX, user_id)
     interval = amo.EDITOR_VIEWING_INTERVAL
 
     # Check who is viewing.
     currently_viewing = cache.get(key)
 
     # If nobody is viewing or current user is, set current user as viewing
     if not currently_viewing or currently_viewing == user_id:
-        # We want to save it for twice as long as the ping interval,
-        # just to account for latency and the like.
-        cache.set(key, user_id, interval * 2)
-        currently_viewing = user_id
-        current_name = request.user.name
-        is_user = 1
+        # Get a list of all the reviews this user is locked on, but first
+        # filter for expired reviews that aren't in the cache.
+        review_locks = filter(lambda r: cache.get(r), cache.get(user_key, {}))
```

Comment:
Not sure how many entries we're expecting here but you could utilize `cache.get_many` here for a better experience.